tests/common/dnspolicy/dnspolicy_controller_single_cluster_test.go

```diff
@@ -115,15 +115,24 @@ var _ = Describe("DNSPolicy Single Cluster", func() {
 		if dnsPolicy != nil {
 			err := k8sClient.Delete(ctx, dnsPolicy)
 			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
-			// Wait until dns records are finished deleting since it can't finish deleting without the DNS provider secret
-			Eventually(func(g Gomega) {
-				dnsRecords := &kuadrantdnsv1alpha1.DNSRecordList{}
-				err := k8sClient.List(ctx, dnsRecords, client.InNamespace(testNamespace))
-				g.Expect(err).NotTo(HaveOccurred())
-				g.Expect(dnsRecords.Items).To(HaveLen(0))
-			}).WithContext(ctx).Should(Succeed())
+		}
 
+		//Ensure all dns records in the test namespace are deleted
+		dnsRecords := &kuadrantdnsv1alpha1.DNSRecordList{}
+		err := k8sClient.List(ctx, dnsRecords, client.InNamespace(testNamespace))
+		Expect(err).ToNot(HaveOccurred())
+		for _, record := range dnsRecords.Items {
+			err := k8sClient.Delete(ctx, &record, client.PropagationPolicy(metav1.DeletePropagationForeground))
+			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
```

Comment:
I might be missing some context here, but I guess this can mask an issue if a DNSPolicy related DNSRecord was not actually marked for deletion when the DNSPolicy is marked for deletion also, since it's deleted by the test also 🤔 

This seems fine if say a DNSRecord resource was created by itself without going through a DNSPolicy or if the DNSPolicy no longer manages the DNSRecord deletion 🤔 Is this the intention / why we want to do this?

Reply:
Some of the tests create different policies, if they aren't dealt with in the tests themselves or in a different AfterEach, then they aren't ever marked for deletion and will likely block the deletion of the namespace. Since this is really just trying to cleanup (its not trying to assert deletion logic works, we have explicit tests for that), the safest thing to do is just make sure all DNSRecord resources are deleted, even though ideally there should be none needing it by this point.

Reply:
Oh I see, that's fair then 👍